plugins/modules/ec2_vpc_nat_gateway_info.py

```diff
@@ -69,51 +68,126 @@
+                returned: always
+                type: str
+                sample: eipalloc-0853e66a40803da76
+            network_interface_id:
+                description: The ID of the network interface associated with the NAT gateway
+                returned: always
+                type: str
+                sample: eni-0a37acdbe306c661c
+            private_ip:
+                description: The private IP address associated with the Elastic IP address
+                returned: always
+                type: str
+                sample: 10.0.238.227
+            public_ip:
+                description: The Elastic IP address associated with the NAT gateway
+                returned: always
+                type: str
+                sample: 34.204.123.52
+    nat_gateway_id:
+        description: The ID of the NAT gateway
+        returned: always
+        type: str
+        sample: nat-0c242a2397acf6173
+    state:
+        description: state of the NAT gateway
+        returned: always
+        type: str
+        sample: available
+    subnet_id:
+        description: The ID of the subnet in which the NAT gateway is located
+        returned: always
+        type: str
+        sample: subnet-098c447465d4344f9
+    vpc_id:
+        description: The ID of the VPC in which the NAT gateway is located
+        returned: always
+        type: str
+        sample: vpc-02f37f48438ab7d4c
+    tags:
+        description: Tags applied to the NAT gateway
+        returned: always
+        type: dict
+        sample:
+            Tag1: tag1
+            Tag_2: tag_2
 '''
 
-import json
 
 try:
     import botocore
 except ImportError:
     pass  # Handled by AnsibleAWSModule
 
-from ansible.module_utils._text import to_native
 from ansible_collections.amazon.aws.plugins.module_utils.ec2 import AWSRetry
 from ansible_collections.amazon.aws.plugins.module_utils.core import AnsibleAWSModule
 from ansible_collections.amazon.aws.plugins.module_utils.ec2 import camel_dict_to_snake_dict
 from ansible_collections.amazon.aws.plugins.module_utils.ec2 import ansible_dict_to_boto3_filter_list
 from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
+from ansible_collections.amazon.aws.plugins.module_utils.core import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.core import normalize_boto3_result
 
 
-def date_handler(obj):
-    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
+@AWSRetry.jittered_backoff(retries=10)
+def _describe_nat_gateways(client, **params):
+    try:
+        paginator = client.get_paginator('describe_nat_gateways')
+        return paginator.paginate(**params).build_full_result()['NatGateways']
+    except is_boto3_error_code('InvalidNatGatewayID.NotFound'):
+        return list()
+    except is_boto3_error_code('NatGatewayMalformed'):  # pylint: disable=duplicate-except
```

Comment:
In this case we should exit and send the error message to the user instead of an empty list; maybe they typoed the gateway ID in their playbook and it could be hard to troubleshoot what's happening if we don't tell them this is a malformed ID.

Reply:
@jillr I added added exit_json with the error message. Thank you.

Reply:
This should be module.fail_json_aws, so that a task with a malformed ID will fail and the user's playbook would not continue.  Passing a malformed gateway ID would be an error.

Reply:
@jillr, you're right. Just changed. Thank you.